### PR TITLE
Retroactively enable ZIP 216 before NU5 activation

### PR DESCRIPTION
**Pull request by str4d**
This completes the work started in zcash/zcash#6000.

Closes zcash/zcash#6396.

### CODE REVIEW

File: src/rust/src/rustzcash.rs [hunk 1]
```diff
@@ -407,7 +401,7 @@ pub extern "C" fn librustzcash_sapling_compute_nf(
 ) -> bool {
     // ZIP 216: Nullifier derivation is not consensus-critical
     // (nullifiers are revealed, not calculated by consensus).
```
Comment (daira) [inline src/rust/src/rustzcash.rs:403]:
```suggestion
    // ZIP 216: Nullifier derivation is not consensus-critical
    // (nullifiers are revealed, not calculated by consensus).
    // In any case, ZIP 216 is now enabled retroactively.
```

**Review by daira — DISMISSED** [review]
utACK with non-blocking suggestions (modulo test failures).

**Comment by str4d** [comment]
[Rebased on `master`](https://github.com/zcash/zcash/compare/5af383d10bd4f312e8db2eabc14df7d215d0b03e..05045949c583852a5f3efb91b45858dc09c8d845) to include v5.4.0-rc4 changes and fix tests, and then [force-pushed](https://github.com/zcash/zcash/compare/05045949c583852a5f3efb91b45858dc09c8d845..9ce6753909057ba13dabc3c11cca826a6c4e3e77) to address @daira's comments.

**Review by nuttycom — APPROVED** [review]
utACK

**Comment by str4d** [comment]
I ran a full reindex with this branch, and was able to reindex up to and past NU5 without errors, confirming that ZIP 216 is compatible with the revealed coinbase outputs.

**Review by daira — APPROVED** [review]
utACK 9ce6753909057ba13dabc3c11cca826a6c4e3e77

**Comment by daira** [comment]
The changes in this PR are correct, but I want to do another review pass over the resulting code to make sure we are completely consistent with #672.

**Review by daira — APPROVED** [review]
Yes, this can be merged now.

**Pull request merged**